include/cvc5/cvc5_proof_rule.h

```diff
@@ -1468,24 +1468,27 @@ enum ENUM(ProofRule) : uint32_t
    * .. math::
    *
    *   \inferruleSC{(t_1\cdot t_2) = (s_1 \cdot s_2),\,
-   *   \mathit{len}(t_1) \neq \mathit{len}(s_1)\mid b}{(t_1 = s_1\cdot r_t)
-   *   \vee (s_1 = t_1\cdot r_s)}{if $b=\bot$}
+   *   \mathit{len}(t_1) \neq \mathit{len}(s_1)\mid b}{((t_1 = s_1\cdot r)
+   *   \vee (s_1 = t_1\cdot r)) \wedge r \neq \epsilon \wedge \mathit{len}(r)>0}{if $b=\bot$}
```

Comment:
Isn't it redundant to add that it both has a length greater than zero and that it's different from the empty string? Or this is to save the trouble of having to do an inference for one of these cases?

Reply:
Yes, it short circuits having to do a separate inference.  The strings solver can be sensitive to these details.